src/main/java/org/commcare/formplayer/configuration/WebSecurityConfig.java

```diff
@@ -0,0 +1,84 @@
+package org.commcare.formplayer.configuration;
+
+import org.commcare.formplayer.auth.CommCareSessionAuthFilter;
+import org.commcare.formplayer.auth.HmacAuthFilter;
+import org.commcare.formplayer.services.FormSessionService;
+import org.commcare.formplayer.services.HqUserDetailsService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.config.annotation.authentication.builders.AuthenticationManagerBuilder;
+import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
+import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter;
+import org.springframework.security.config.http.SessionCreationPolicy;
+import org.springframework.security.web.authentication.UsernamePasswordAuthenticationFilter;
+import org.springframework.security.web.authentication.preauth.PreAuthenticatedAuthenticationProvider;
+
+@Configuration
+@EnableWebSecurity
+public class WebSecurityConfig extends WebSecurityConfigurerAdapter {
+    @Autowired
+    private HqUserDetailsService userDetailsService;
+
+    @Autowired
+    private FormSessionService formSessionService;
+
+    @Value("${commcarehq.formplayerAuthKey}")
+    private String formplayerAuthKey;
+
+    @Override
+    protected void configure(HttpSecurity http) throws Exception {
+        disableDefaults(http);
+
+        http
+            .authorizeRequests()
+            .antMatchers("/serverup", "/validate_form", "/favicon.ico")
+            .permitAll();
+        http.authorizeRequests().antMatchers("/**").authenticated();
+        http.addFilterAt(getHmacAuthFilter(), UsernamePasswordAuthenticationFilter.class);
+        http.addFilterAfter(sessionAuthFilter(), HmacAuthFilter.class);
+        http.cors();
+    }
+
+    private HmacAuthFilter getHmacAuthFilter() {
+        return HmacAuthFilter.builder()
+                .hmacKey(formplayerAuthKey)
+                .formSessionService(formSessionService)
+                .build();
+    }
+
+    private void disableDefaults(HttpSecurity http) throws Exception {
+        http.csrf().disable();
```

Comment:
Is this matching our existing defaults somehow? I didn't see an obvious place where it was being copied from the old code config.

Reply:
We didn't have any CSRF before. If we want to enable it we'd need to implement it on this side as well as the UI since there's currently no support for getting or sending CSRF tokens in Web Apps.

Reply:
https://dimagi-dev.atlassian.net/browse/USH-827

Reply:
Got it, thanks for clarifying!